src/client/cli/cmd/disable_zones.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "disable_zones.h"
+
+#include "animated_spinner.h"
+#include "common_callbacks.h"
+#include "common_cli.h"
+#include "multipass/cli/argparser.h"
+#include "multipass/cli/client_common.h"
+
+#include <QCommandLineOption>
+
+namespace multipass::cmd
+{
+ReturnCode DisableZones::run(ArgParser* parser)
+{
+    if (const auto ret = parse_args(parser); ret != ParseCode::Ok)
+        return parser->returnCodeFrom(ret);
+
+    if (ask_for_confirmation)
+    {
+        const PlainPrompter prompter{term};
+        auto answer = prompter.prompt("This operation will forcefully stop the VMs in zone3. Proceed? (Yes/No)");
+        while (!std::regex_match(answer, client::yes_answer) && !std::regex_match(answer, client::no_answer))
+            answer = prompter.prompt("Please answer (Yes/No)");
+
+        if (std::regex_match(answer, client::no_answer))
+            return Ok;
+    }
+
+    AnimatedSpinner spinner{cout};
+    spinner.start(format("Disabling {}", fmt::join(zone_names, ", ")));
+
+    ZonesStateRequest request{};
+    for (const auto& zone_name : zone_names)
+        request.add_zones(zone_name);
+    request.set_available(false);
+    request.set_verbosity_level(parser->verbosityLevel());
+
+    const auto on_success = [&](const ZonesStateReply&) {
+        spinner.stop();
+        cout << (zone_names.size() == 1 ? "Zone" : "Zones") << " disabled successfully" << std::endl;
```

Comment:
We can also use something like `<N> zone(s) disabled successfully.`, so the user can verify that it matches the amount of zones requested.